### PR TITLE
Test bumping dune on CI.

### PR DESCRIPTION
**Pull request by cristianoc**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -114,7 +114,7 @@ jobs:
           command: |
             opam install reason.3.6.0 dune.1.11.4 --yes | cat
```
Comment (Et7f3) [inline .circleci/config.yml:115]: need to be bumped here.

**Comment by cristianoc** [comment]
On win:
```
[NOTE] Package reason is already installed (current version is 3.6.0).
The following dependencies couldn't be met:
  - dune -> ocaml >= 4.07
```

**Comment by Et7f3** [comment]
try a `opam update`

**Comment by Et7f3** [comment]
If you want you can give me right on this branch.

**Comment by cristianoc** [comment]
Might revisit later if there's a compelling advantage. No reason to dig more at the moment.

**Comment by Et7f3** [comment]
Ok we can't solve it because of: https://github.com/fdopen/opam-repository-mingw/issues/75 and latest version of dune in this repo is 2.0.0. We can move to esy to unblock this. If you don't mind changing from Circle CI to GHA you can have ci already setup of hello-reason.

**Pull request closed without merging**